src/valid/function.rs

```diff
@@ -363,11 +363,17 @@ impl super::Validator {
                 .into_other());
         }
         match context.expressions[result] {
-            //TODO: support atomic result with comparison
+            //TODO: does the result of an atomicCompareExchange need additional validation, or does the existing validation for
+            // the struct type it returns suffice?
```

Comment:
We should still validate that `comparison_ty.old_value.ty.kind == ptr_kind && comparison_ty.old_value.ty.width == ptr_width` (pseudocode).

Reply:
`validate_function` and `validate_expression` now both call into `validate_atomic_compare_exchange_struct`, which checks the struct members' names and types (including checking that `old_value` satisfies the same predicate on scalars that the functions previously checked directly for).